### DISCUSSION TITLE
Are there plans to revamp the overall app architecture to adopt android guidlines on seperation of data and ui layer?

### DISCUSSION THREAD

**Discussion by HolonProduction** [FossifyOrg/General-Discussion#219] [General]
I was trying to fix some issues I have with the music player app, but given that the apps architecture differs quite a bit from the ["textbook" android architecture](https://developer.android.com/topic/architecture) it is a bit hard for me to get into it. In particular I am missing a clear separation of data and ui layer. What's the projects stance on this?

For reference one issue that results from the missing separation:
- https://github.com/FossifyOrg/Music-Player/issues/68

**Comment by naveensingh** [comment]
> Are there plans to revamp the overall app architecture to adopt android guidlines on seperation of data and ui layer? 

Yes but not anytime soon.
